File: src/chip8/sound.rs
pub struct Sound {
    value: u8,
}

impl Sound {
    pub fn new() -> Self {
        Self { value: 0 }
    }

    pub fn write(&mut self, value: u8) {
        self.value = value;
    }

    pub fn step(&mut self) {
        if self.value > 0 {
            self.value -= 1;
        }
    }
}

File: chip8/sound.go
package chip8

type Sound struct {
	value uint8
}

func NewSound() *Sound {
	return &Sound{}
}

func (s *Sound) Write(value uint8) {
	s.value = value
}

func (s *Sound) Step() {
	if s.value > 0 {
		s.value--
	}
}
